### PR TITLE
PRL-4195:Respondent C8 generation while amending respondent details

### PR DESCRIPTION
**Pull request by khurshedk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-4191
https://tools.hmcts.net/jira/browse/PRL-4195
https://tools.hmcts.net/jira/browse/PRL-4376
https://tools.hmcts.net/jira/browse/PRL-4193

### Change description ###
Respondent C8 generation while amending confidential details of respondent for C100/FL401 case.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/models/dto/ccd/CaseData.java [hunk 1]
```diff
@@ -753,17 +753,8 @@ public CaseData setIssueDate() {
...
-    @JsonProperty("respondentAc8Documents")
-    private List<Element<ResponseDocuments>> respondentAc8Documents;
-    @JsonProperty("respondentBc8Documents")
-    private List<Element<ResponseDocuments>> respondentBc8Documents;
-    @JsonProperty("respondentCc8Documents")
-    private List<Element<ResponseDocuments>> respondentCc8Documents;
-    @JsonProperty("respondentDc8Documents")
-    private List<Element<ResponseDocuments>> respondentDc8Documents;
-    @JsonProperty("respondentEc8Documents")
-    private List<Element<ResponseDocuments>> respondentEc8Documents;
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/models/dto/ccd/CaseData.java:765]: we need to double check on data migration for this. once we remove this fields, any case with this data present will fail. so we must verify

File: src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java [hunk 2]
```diff
@@ -248,92 +253,157 @@ private void generateC8DocumentsForRespondents(Map<String, Object> updatedCaseDa
...
     }
 
-    private  void populateC8Documents(Map<String, Object> updatedCaseData, CaseData caseData,
-                                      Document c8FinalDocument,
-                                      Document c8WelshDocument, int partyIndex) {
-        if (null != c8FinalDocument && partyIndex >= 0) {
-            ResponseDocuments c8ResponseDocuments = ResponseDocuments.builder()
-                .dateTimeCreated(LocalDateTime.now())
-                .build();
+    public Boolean checkIfDetailsChanged(CallbackRequest callbackRequest, Element<PartyDetails> respondent) {
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java:266]: change the method name please.

File: src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java [hunk 3]
```diff
@@ -248,92 +253,157 @@ private void generateC8DocumentsForRespondents(Map<String, Object> updatedCaseDa
...
+                                                                       Boolean isDetailsChanged,
+                                                                       Element<PartyDetails> respondent)
+        throws  Exception {
+        Document c8FinalDocument;
+        Document c8FinalWelshDocument;
+        String partyName = respondent.getValue().getLabelForDynamicList();
+        if (dataMap.containsKey(IS_CONFIDENTIAL_DATA_PRESENT)) {
+            if ((isDetailsChanged
+                || CollectionUtils.isEmpty(c8Documents))) {
+                String fileName = C_8_OF + partyName
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java:367]: Is it C_8_OF or C8_OF?

File: src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java [hunk 4]
```diff
@@ -248,92 +253,157 @@ private void generateC8DocumentsForRespondents(Map<String, Object> updatedCaseDa
...
+                    + " " + LocalDateTime.now(ZoneId.of(LONDON_TIME_ZONE)).format(dateTimeFormatter);
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java:368]: Time format is inline with the standard, e.g. 20 Oct 2023

File: src/main/java/uk/gov/hmcts/reform/prl/models/dto/ccd/RespondentC8Document.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Builder;
+import lombok.Data;
+import uk.gov.hmcts.reform.prl.models.Element;
+import uk.gov.hmcts.reform.prl.models.complextypes.citizen.documents.ResponseDocuments;
+
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@Builder(toBuilder = true)
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/models/dto/ccd/RespondentC8Document.java:14]: Include @JsonInclude(JsonInclude.Include.NON_NULL)

**Review by AlokDatta — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/prl/models/dto/ccd/RespondentC8Document.java (hunk 5 above)
Comment (JamesCutts00) [reply src/main/java/uk/gov/hmcts/reform/prl/models/dto/ccd/RespondentC8Document.java:15]: added the above

File: src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java (hunk 2 above)
Comment (JamesCutts00) [reply src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java]: changed to checkIfConfidentialityDetailsChangedRespondent

**Review by sahaSonali — APPROVED** [review]

**Pull request merged**